### ISSUE TITLE
Is there any limit if using Google Plus?

### ISSUE THREAD

**Issue by aalfiann** [japnimedev/Google-Plus-Stream#2]
Because it use Google Photos maybe there is a limit..

**Comment by Amrinder-Ghumman** [comment]
I believe there are none because it's the social media platform where a video could get millions of views. so I think there are none.

**Comment by Amrinder-Ghumman** [comment]
I believe there are none because it's the social media platform where a video could get millions of views. so I think there are none.

**Comment by aalfiann** [comment]
Yes there is no limit if you use their platform like Google API. But if not there is limit. If same ip address has high request then you will get header 429 or too many request.

**Comment by Yokutto** [comment]
@Amrinder-Ghumman Google Plus videos are served by picasa (google photos) and both have limitations

**Comment by Amrinder-Ghumman** [comment]
@Yokutto  i think you didn't understand what I'm trying to tell you here. pls, let me explain. you see if you upload a video to facebook and it can be seen as many times as it could be so there are no limitations because its a social media platform and these platforms are meant to share so if there would be a limitation then no one will going to use that social media platform. all of the above I mention goes same for Google Plus. it is also a social media platform. think it in this way if there will be a limit then when official google uploads a video to Google Plus and share and if it reaches 1 Million people what do you think will happens, i bet the video will still play the same. so this is what I'm trying to tell you, i know what you mean but remember this before fetching the video from cloud some parameters also passes with the request to the file which determines the file life. so i believe there are no limits here. not for Google Plus, I suppose.

**Comment by Yokutto** [comment]
@Amrinder-Ghumman well... you have a link for video with various views? i used google plus in high traffic website and video get deleted from gplus(https://plus.google.com/photos/photo/113826803239731078195/6546903128219140482 - 404 error)

facebook also have limitations, only 720p and 240p with "no_ratelimit" in html source have no limits

**Comment by Amrinder-Ghumman** [comment]
@Yokutto you see their are two things you can do to prevent you video from getting delete, i believe your video was deleted because you Source url was visible, try looking into base_64 encoding and decoding. And secondly try adding a cache function to the script i have added base_64 encryption but was not able to add cache function. As far as I'm concerned you video got deleted for copyrighted material and was visible to others. If not then can you pm me with some more details about how much views did it get and other stuff. Well you know i could be wrong about the social Media platform stuff but it will be unlikely.

**Comment by Yokutto** [comment]
@Amrinder-Ghumman Maybe there is some way to get unlimited streaming links, but it probably will not be through the source code of the page. All videos served by google with "ipbits=8" in the link end up being erased by limit of view, "ipbits=0" used in google drive/youtube for example has no limit (at least in my tests). i used google plus links with 500/700 viewers at the same time, the script have cache links (this not save bandwidth). all the videos that take a lot of visualization in google plus are hosted on youtube, I could not find any video that is not pornography with much traffic
